core/src/subgraph/instance_manager.rs

```diff
@@ -322,16 +328,28 @@ where
         })
         // Process blocks from the stream as long as no restart is needed
         .fold(ctx, move |ctx, block| {
+            let needs_restart_set = needs_restart_set.clone();
```

Comment:
This declaration looks dead.

Reply:
I think it's needed to get the ownership moves right with the two nested closures.

Reply:
_maybe_ you can drop the `.clone()`/

Reply:
I think I just tried all possible combinations of not having the extra assignment or having it, with and without cloning the outer var or the inner var or both... nope :-/

Reply:
Then this is it, thanks for trying it out.